.github/actions/setup-java/action.yml

```diff
@@ -0,0 +1,37 @@
+name: "Setup JDK"
+description: "setup JDK and gradle caching"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Validate Gradle Wrapper
+      uses: gradle/actions/wrapper-validation@v4
+
+    - name: Copy CI gradle.properties
+      shell: bash
+      run: |
+        mkdir -p ~/.gradle
+        cp .github/ci-gradle.properties ~/.gradle/gradle.properties
+
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'temurin'
+        java-version: '17'
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v4
+      with:
+        gradle-version: wrapper
+
+    - name: set konan cache key
+      id: konan-cache-key
+      shell: bash
+      run: echo "KOTLIN_VERSION=$(grep -oE 'kotlin\s*=\s*"[0-9.]*"' gradle/libs.versions.toml | grep -oE '[0-9.]+')" >> $GITHUB_OUTPUT
+
+    - name: cache Konan
```

Comment:
@l2hyunwoo Konan? 살펴볼 수 있는 레퍼런스가 있을까용

Reply:
이건 [카이기것도](https://github.com/DroidKaigi/conference-app-2024/blob/main/.github/workflows/ios-build-test.yml) 참고해보고 [github action cache 예제](https://github.com/rnett/github-actions-gradle-cache)도 참고해봤어요.

Konan 쓴 이유는 Kotlin/Native cache를 지원해서 사용한거긴 합니다.

> It comes with defaults set for Gradle, so by default it cashes these directories in ~/.gradle:
> 
> nodejs
> wrapper
> yarn
> jdks
> children of caches expect for build-cache*, to avoid duplicating.
> **It also caches ~/konan, for Kotlin/Native support.**